build_tools/python/benchmark_suites/iree/x86_64_benchmarks.py

```diff
@@ -40,26 +40,44 @@ def _generate_default(
       self, device_specs: List[common_definitions.DeviceSpec]
   ) -> Tuple[List[iree_definitions.ModuleGenerationConfig],
              List[iree_definitions.E2EModelRunConfig]]:
-    gen_configs = [
+    # Create benchmarks for small workloads.
+    gen_configs_small = [
         iree_definitions.ModuleGenerationConfig.build(
             compile_config=self.CASCADELAKE_COMPILE_CONFIG,
             imported_model=iree_definitions.ImportedModel.from_model(model))
-        for model in model_groups.SMALL + model_groups.LARGE
+        for model in model_groups.SMALL
     ]
-
-    default_execution_configs = [
-        module_execution_configs.ELF_LOCAL_SYNC_CONFIG
-    ] + [
-        module_execution_configs.get_elf_local_task_config(thread_num)
-        for thread_num in [1, 4, 8]
+    execution_configs_small = [
+        module_execution_configs.get_elf_local_task_config(1),
+        module_execution_configs.get_elf_local_task_config(8)
     ]
+    run_configs_small = benchmark_suites.iree.utils.generate_e2e_model_run_configs(
+        module_generation_configs=gen_configs_small,
+        module_execution_configs=execution_configs_small,
+        device_specs=device_specs)
 
-    run_configs = benchmark_suites.iree.utils.generate_e2e_model_run_configs(
-        module_generation_configs=gen_configs,
-        module_execution_configs=default_execution_configs,
+    # Create benchmarks for large workloads.
+    gen_configs_large = [
+        iree_definitions.ModuleGenerationConfig.build(
+            compile_config=self.CASCADELAKE_COMPILE_CONFIG,
+            imported_model=iree_definitions.ImportedModel.from_model(model))
+        for model in model_groups.LARGE
+    ]
+    # We use higher thread counts for large workloads.
+    execution_configs_large = [
+        module_execution_configs.get_elf_local_task_config(1),
+        module_execution_configs.get_elf_local_task_config(8),
+        module_execution_configs.get_elf_local_task_config(16),
+        # We use (MAX_CORE - 2) here to determine the maximum number of threads to use on a NUMA node.
+        module_execution_configs.get_elf_local_task_config(28),
```

Comment:
Currently we are still using c2-standard-16 to run benchmarks, so these configurations might not produce meaningful results for now. Can we delay these until we switch to c2-standard-60?

Also do we want to test 28 threads on c2-standard-60? The single numa node only has 15cores, so this is using hyper-threading, which from my outdated information, only provides ~20% of extra performance.

Reply:
Ah gotcha. So c2-standard-60 has 15 cores on a single numa node and without hyperthreading, we would use at most 15 threads. Also from Anush's experiments, it sounds like we should really be using max 13 threads instead of 16.

Reply:
Updated to 13 threads and removed 28 threads.